rootfs/apply.go

```diff
@@ -112,18 +112,27 @@ func ApplyLayerWithOpts(ctx context.Context, layer Layer, chain []digest.Digest,
 
 func applyLayers(ctx context.Context, layers []Layer, chain []digest.Digest, sn snapshots.Snapshotter, a diff.Applier, opts []snapshots.Opt, applyOpts []diff.ApplyOpt) error {
 	var (
-		parent  = identity.ChainID(chain[:len(chain)-1])
-		chainID = identity.ChainID(chain)
-		layer   = layers[len(layers)-1]
-		diff    ocispec.Descriptor
-		key     string
-		mounts  []mount.Mount
-		err     error
+		parent   = identity.ChainID(chain[:len(chain)-1])
+		chainID  = identity.ChainID(chain)
+		chainStr = chainID.String()
+		layer    = layers[len(layers)-1]
+		diff     ocispec.Descriptor
+		key      string
+		mounts   []mount.Mount
+		err      error
 	)
 
 	for {
 		key = fmt.Sprintf(snapshots.UnpackKeyFormat, uniquePart(), chainID)
 
+		// inherits annotations which are provided as snapshot labels.
+		labels := snapshots.FilterInheritedLabels(diff.Annotations)
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels["containerd.io/snapshot.ref"] = chainStr
+		opts = append(opts, snapshots.WithLabels(labels))
```

Comment:
This shouldn't be added in the for loop unless creating a new `opts` variable.

Perhaps we need a `snapshots.WithLabel` to append a single label at some point. The way this is written up along with the implementation in the unpack package unnecessarily add this ref label to the committed snapshot as well.

Reply:
Would just moving it out of the for loop and just do it once suffice? Would a `Stat()` before the `Prepare()` avoid applying the label to a committed snapshot?